zerver/tests/test_messages.py

```diff
@@ -2492,7 +2492,7 @@ def check_message(self, msg_id: int, topic_name: Optional[str]=None,
                       content: Optional[str]=None) -> Message:
         msg = Message.objects.get(id=msg_id)
         cached = MessageDict.wide_dict(msg)
-        MessageDict.finalize_payload(cached, apply_markdown=False, client_gravatar=False)
+        cached = MessageDict.finalize_payload(cached, apply_markdown=False, client_gravatar=False)
```

Comment:
This might be cleaner if you just pass `MessageDict.wide_dict(msg)` to `finalize_payload`, rather than reassigning `cached`.

Reply:
Okay

Reply:
You don't want to call the finalized version `cached` here, @timabbott, because it's not the value from the cache.

I would use these locals for clarity:

* cached_dict
* finalize_dict
* uncached_dict

I suggest just copy/paste-ing this (and fix any typos):

~~~
2495         # First see what happens when we take the cached wide
2496         # dictionary and then produce a finalized copy.
2497         cached_dict = MessageDict.wide_dict(msg)
2498         finalized_dict = MessageDict.finalize_payload(cached_dict, apply_markdown=False, client_gravatar=False)
2499
2500         # And we'll compare that to an actual DB lookup that
2501         # gets post-processed.
2502         uncached_dict = MessageDict.to_dict_uncached_helper(msg)
2503         MessageDict.post_process_dicts([uncached_dict], apply_markdown=False, client_gravatar=False)
2504
2505         self.assertEqual(finalized_dict, uncached_dict)

Reply:
Actually, the `cached` name makes zero sense here, because the first value is also taken from the database.  Hrm.

Reply:
Wow, this misleading name goes all the way back to 2013 (6b820038ad6c0bdef23eee45a38c7f0e676a91c3).  

Here is the original version:

~~~ py
1896 class EditMessageTest(AuthedTestCase):
1897     def check_message(self, msg_id, subject=None, content=None):
1898         msg = Message.objects.get(id=msg_id)
1899         cached = msg.to_dict(False)
1900         uncached = msg.to_dict_uncached(False)
1901         self.assertEqual(cached, uncached)
1902         if subject:
1903             self.assertEqual(msg.subject, subject)
1904         if content:
1905             self.assertEqual(msg.content, content)
1906         return msg
~~~

Back then the caching code was kind of strange, because we'd actually **have** a Message object and then use a cache to avoid re-serializing it.  I think that's changed significantly over time, which is why the current test is just confusing now.

~~~
 697     def to_dict(self, apply_markdown):
 698         return extract_message_dict(self.to_dict_json(apply_markdown))
 699
 700     @cache_with_key(to_dict_cache_key, timeout=3600*24)
 701     def to_dict_json(self, apply_markdown):
 702         return stringify_message_dict(self.to_dict_uncached(apply_markdown))
~~~

Reply:
I'll write a quick PR now to clean this up.